src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -817,6 +818,16 @@ def _resolve_extension_override_warning(self, cmd):  # pylint: disable=no-self-u
         if isinstance(cmd.command_source, ExtensionCommandSource) and cmd.command_source.overrides_command:
             logger.warning(cmd.command_source.get_command_warn_msg())
 
+    def _resolve_upcoming_breaking_change_warning(self, cmd):
+        if not self.cli_ctx.only_show_errors:
+            from ..breaking_change import iter_command_breaking_changes, BreakingChange
+
+            for bc in iter_command_breaking_changes(cmd.name):
+                if isinstance(bc, str):
+                    print(bc, file=sys.stderr)
```

Comment:
I would like to confirm the reason why we prefer to use `print` over `log.warning`

Reply:
I have changed this to `logger.warning`. The main reason to use `print` previously is to follow the code in `_resolve_preview_and_deprecation_warnings()` method. However, the deprecate_info message is a ColorizedString which is defined in `knack.util.py`. And I think it is not suitable to use it directly.